lib/stat_generator.rb

```diff
@@ -159,19 +171,71 @@ def total_goals_by_team(team_id)
     end
   end
 
+  def total_goals_by_team_when_away(team_id)
+    @game_teams.sum do |game_team|
+      if game_team.team_id == team_id && game_team.home_or_away == "away"
+        game_team.goals
+      else
+        0
+      end
+    end
+  end
+
+  def total_goals_by_team_when_home(team_id)
+    @game_teams.sum do |game_team|
+      if game_team.team_id == team_id && game_team.home_or_away == "home"
+        game_team.goals
+      else
+        0
+      end
+    end
+  end
+
   def average_goals_per_game_by_team(team_id)
     (total_goals_by_team(team_id).to_f / total_games_played_by_team(team_id)).round(2)
   end
 
+  def average_goals_per_game_by_team_when_away(team_id)
+    (total_goals_by_team_when_away(team_id).to_f / total_games_played_by_team_when_away(team_id)).round(2)
+  end
+
+  def average_goals_per_game_by_team_when_home(team_id)
+    (total_goals_by_team_when_home(team_id).to_f / total_games_played_by_team_when_home(team_id)).round(2)
+  end
+
   def best_offense
     @teams.max_by do |team|
       average_goals_per_game_by_team(team.team_id)
     end.team_name
   end
 
+  def highest_scoring_visitor
+    @teams.max_by do |team|
+      average_goals_per_game_by_team_when_away(team.team_id)
+    end.team_name
+  end
+
+  def highest_scoring_home_team
+    @teams.max_by do |team|
+      average_goals_per_game_by_team_when_home(team.team_id)
+    end.team_name
+  end
+
   def worst_offense
     @teams.min_by do |team|
       average_goals_per_game_by_team(team.team_id)
     end.team_name
   end
+
+  def lowest_scoring_visitor
+    @teams.min_by do |team|
+      average_goals_per_game_by_team_when_away(team.team_id)
```

Comment:
Can we combine those ? the only thing changing in the "home" and the "away" right?

Reply:
Overall Great job!!